### PR TITLE
sidecarSet support namespace selector inject

### PR DESCRIPTION
**Pull request by hantmac**
Signed-off-by: hantmac <hantmac@outlook.com>

### Ⅰ. Describe what this PR does

SidecarSet support namespace selector pod inject.

### Ⅱ. Does this pull request fix one issue?

Fix  https://github.com/openkruise/kruise/issues/948

### CODE REVIEW

File: pkg/control/sidecarcontrol/util.go [hunk 1]
```diff
@@ -72,9 +73,10 @@ type SidecarSetUpgradeSpec struct {
 }
 
 // PodMatchSidecarSet determines if pod match Selector of sidecar.
-func PodMatchedSidecarSet(pod *corev1.Pod, sidecarSet appsv1alpha1.SidecarSet) (bool, error) {
-	//If matchedNamespace is not empty, sidecarSet will only match the pods in the namespace
-	if sidecarSet.Spec.Namespace != "" && sidecarSet.Spec.Namespace != pod.Namespace {
+func PodMatchedSidecarSet(cl client.Client, pod *corev1.Pod, sidecarSet appsv1alpha1.SidecarSet) (bool, error) {
+	//If pod's namespace in selectedNamespaces, sidecarSet will match the pods in the selectedNamespaces
+	selectedNamespaces := GetNamespacesFromNamespaceSelector(cl, sidecarSet.Spec)
```
Comment (veophi) [inline pkg/control/sidecarcontrol/util.go:78]: Is it better to use the following methods ?
```go
selector, err := metav1.LabelSelectorAsSelector(sidecarSet.Spec.NamespaceSelector)
selector.Matches(pod.Namespace)
```

Comment (hantmac) [reply pkg/control/sidecarcontrol/util.go:78]: `Matches` only support `Matches(label)` such as `Matches(pod.Labels)`, not support Matches(pod.Namespace) 😢

Comment (veophi) [reply pkg/control/sidecarcontrol/util.go:78]: > 
Oh, Sorry... 
Maybe, we can just `Get` this namespace of pod instead of `List` ?

Comment (hantmac) [reply pkg/control/sidecarcontrol/util.go:78]: `selectedNamespaces ` is a  slice so we neeed to list them 🤔

**Pull request closed without merging**